private/numerics/functions/elliptic.rkt

```diff
@@ -0,0 +1,372 @@
+  [Carlson-elliptic-2 three-nonnegative-reals->number]
+
+  ;; Elliptic integrals of the first kind: F(φ,k)
+
+  ;; TODO: does it make sense to allow k²sin²φ > 1? NO
+  ;;       first param should be in [0, 2π]
+  [elliptic-integral-F (-> real?
+                           (and/c real? positive? (</c 1))
+                           number?)]
+
+  [elliptic-integral-E (-> real?
+                           (and/c real? positive? (</c 1))
+                           number?)]
+  
+  [complete-elliptic-integral-K (-> (and/c real? positive? (</c 1))
+                                    number?)]
+  
+  [complete-elliptic-integral-E (-> (and/c real? positive? (</c 1))
+                                    number?)]
+  
+  [elliptic-integrals (-> (and/c real? positive? (</c 1))
+                          (-> flonum? flonum? any)
+                          any)]
+  
+  [first-elliptic-integral (-> (and/c real? positive? (</c 1))
+                               number?)]
+  
+  [second-elliptic-integral (-> (and/c real? positive? (</c 1))
+                                number?)]
+  
+  [first-elliptic-integral&derivative (->i ([k (and/c real? (</c 1))]
+                                            [cont (-> number? number? any)])
+                                           [result (k)
+                                                   (if (= k 0)
+                                                       (cons/c flonum? flonum?)
+                                                       any/c)])]
+  
+  [Jacobi-elliptic-functions (-> number?
+                                 (and/c real? positive? (</c 1))
+                                 (-> number? number? number? any)
+                                 any)]))
+
+(define nonnegative-real? (not/c (and/c real? negative?)))
+
+(define three-nonnegative-reals->number
+  (-> nonnegative-real?
+      nonnegative-real?
+      nonnegative-real?
+      number?))
+
+(require
+ (only-in mechanics π π/2 square machine-ε))
+
+(require
+ (only-in racket/math cosh tanh))
+
+(require
+ (only-in racket/fixnum fx< fx+))
+
+(define (Rf x y z)
+  (define ε (expt machine-ε 1/6))
+  (define C₁ 1/24)
+  (define C₂ 1/10)
+  (define C₃ 3/44)
+  (define C₄ 1/14)
+  (define √x (sqrt x))
+  (define √y (sqrt y))
+  (define √z (sqrt z))
+  (define λₐ (+ (* √x
+                   (+ √y √z))
+                (* √y √z)))
+  (define xp (* 1/4 (+ x λₐ)))
+  (define yp (* 1/4 (+ y λₐ)))
+  (define zp (* 1/4 (+ z λₐ)))
+  (define μ (/ (+ xp yp zp) 3))
+  (define Δx (/ (- μ x) μ))
+  (define Δy (/ (- μ y) μ))
+  (define Δz (/ (- μ z) μ))
+  (define error (max (abs Δx) (abs Δy) (abs Δz)))
+  (if (> error ε)
+      (Rf xp yp zp)
+      (let ([e₂ (- (* Δx Δy)
+                   (* Δz Δz))]
+            [e₃ (* Δx Δy Δz)])
+        (/ (+ 1
+              (* (- (* C₁ e₂)
+                    (+ C₂ (* C₃ e₃)))
+                 e₂)
+              (* C₄ e₃))
+           (sqrt μ)))))
+
+(define Carlson-elliptic₁ Rf)
```

Comment:
I couldn't get `rename-out` to export more than one alias. I export the aliases for backward compatibility.